scene/main/window.cpp

```diff
@@ -1633,7 +1633,10 @@ void Window::popup(const Rect2i &p_screen_rect) {
 
 	if (p_screen_rect != Rect2i()) {
 		set_position(p_screen_rect.position);
-		set_size(p_screen_rect.size);
+		int screen_id = DisplayServer::get_singleton()->window_get_current_screen();
```

Comment:
This will use the size of the screen where the main window is. But a new window might be opening on the other screen (which might have different size). There's `get_screen_from_rect` function that can be used to get screen no. from the window rect.

Reply:
Like this? 

https://github.com/godotengine/godot/blob/b12c2972234918efc2fb859bdd2c4e1be5156595/scene/main/window.cpp#L1636C1-L1636C1